lib/ansible/modules/web_infrastructure/ansible_tower/tower_workflow_template.py

```diff
@@ -0,0 +1,190 @@
+from ansible.module_utils.ansible_tower import (
+    tower_argument_spec,
+    tower_auth_config,
+    tower_check_mode,
+    HAS_TOWER_CLI
+)
+
+try:
+    import tower_cli
+    import tower_cli.exceptions as exc
+    from tower_cli.conf import settings
+except ImportError:
+    pass
+
+
+def main():
+    argument_spec = tower_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True),
+        description=dict(required=False),
+        extra_vars=dict(required=False),
+        organization=dict(required=False),
+        allow_simultaneous=dict(required=False),
+        schema=dict(required=False),
+        survey=dict(required=False),
+        survey_enabled=dict(required=False),
+        state=dict(choices=['present', 'absent'], default='present'),
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+
+    if not HAS_TOWER_CLI:
+        module.fail_json(msg='ansible-tower-cli required for this module')
+
+    name = module.params.get('name')
+    state = module.params.get('state')
+
+    json_output = {'workflow_template': name, 'state': state}
+
+    tower_auth = tower_auth_config(module)
+    with settings.runtime_values(**tower_auth):
+        tower_check_mode(module)
+        wfjt_res = tower_cli.get_resource('workflow')
+        try:
+            params = {}
+            params['name'] = name
+
+            if module.params.get('description'):
+                params['description'] = module.params.get('description')
+
+            if module.params.get('organization'):
+                organization_res = tower_cli.get_resource('organization')
+                try:
+                    organization = organization_res.get(
+                        name=module.params.get('organization'))
+                    params['organization'] = organization['id']
+                except exc.NotFound as excinfo:
+                    module.fail_json(
+                        msg='Failed to update organization source,'
+                        'organization not found: {0}'.format(excinfo),
+                        changed=False
+                    )
+
+            if module.params.get('survey'):
+                params['survey_spec'] = module.params.get('survey')
+
+            for key in ('allow_simultaneous', 'extra_vars', 'survey_enabled',
+                        'description'):
+                if module.params.get(key):
+                    params[key] = module.params.get(key)
+
+            if state == 'present':
+                params['create_on_missing'] = True
+                result = wfjt_res.modify(**params)
+                json_output['id'] = result['id']
+            elif state == 'absent':
+                params['fail_on_missing'] = False
+                result = wfjt_res.delete(**params)
+
+            if module.params.get('schema'):
+                wfjt_res.schema(
+                    wfjt_res.get(name=params['name'])['id'],
```

Comment:
nit-picky optimization, but if the prior step was successful then 'id' should already be in `result`

Also, if the state is "absent", then you need to be sure that you don't hit this code. Either that, or it should throw an error if schema is provided and state is absent.

Reply:
Ok, I've fixed that.